### PR TITLE
improvement/MOBILE-4026 Add session token parameter to syncRealm method in Android SDK

### PR DESCRIPTION
**Pull request by agustk**
Added session token to the realm sync method in the sdk.

### CODE REVIEW

File: sdk/src/main/java/com/meniga/sdk/models/sync/operators/MenigaSyncOperations.java [hunk 1]
```diff
@@ -9,7 +9,7 @@
  */
 public interface MenigaSyncOperations {
 
-	Result<MenigaSync> startSync(Long realmUserId, long timeout);
+	Result<MenigaSync> startSync(Long realmUserId, String sessionToken, long timeout);
```
Comment (arturdm) [inline sdk/src/main/java/com/meniga/sdk/models/sync/operators/MenigaSyncOperations.java]: You can add `@Nullable` to the new parameter.

File: sdk/src/main/java/com/meniga/sdk/models/sync/MenigaSync.kt [hunk 2]
```diff
@@ -196,10 +196,29 @@ data class MenigaSync(
...
+         * Returns a MenigaSync object once the sync procedure has been launched.
+         *
+         * @param realmUserId  The id of the realm account user - a realm is a "department" in e.g. a bank. Most organizations (most often banks) have only one but some
+         * large organizations can have many realms (e.g. insurance, banking, credit cards and so on). This id identifies the user's realm user account.
+         * @param sessionToken A token from a previous (e.g. authentication) operation that should be carried over to the sync process
+         * @param timeout  The amount of time the background process should try to check if the sync has completed before terminating.
+         * @return A new sync object.
+         */
+        @JvmStatic
+        fun syncRealm(realmUserId: Long, sessionToken: String?, timeout: Long, onDone: Interceptor<MenigaSync>? = null): Result<MenigaSync> {
```
Comment (arturdm) [inline sdk/src/main/java/com/meniga/sdk/models/sync/MenigaSync.kt]: There's `MenigaSyncApiTest.kt` file in `integrationTest` where you can put a test case for the method with the session token.

**Review by arturdm — CHANGES_REQUESTED** [review]

File: sdk/src/main/java/com/meniga/sdk/models/sync/operators/MenigaSyncOperations.java (hunk 1 above)
Comment (agustk) [reply sdk/src/main/java/com/meniga/sdk/models/sync/operators/MenigaSyncOperations.java]: Isn't String inherently nullable?

File: sdk/src/main/java/com/meniga/sdk/models/sync/MenigaSync.kt (hunk 2 above)
Comment (agustk) [reply sdk/src/main/java/com/meniga/sdk/models/sync/MenigaSync.kt]: Cool, will see about adding one there.

File: sdk/src/main/java/com/meniga/sdk/models/sync/operators/MenigaSyncOperations.java (hunk 1 above)
Comment (agustk) [reply sdk/src/main/java/com/meniga/sdk/models/sync/operators/MenigaSyncOperations.java]: Sure

File: sdk/src/main/java/com/meniga/sdk/models/sync/operators/MenigaSyncOperationsImp.java [hunk 3]
```diff
@@ -8,12 +8,14 @@
...
 
+import javax.annotation.Nullable;
+
 /**
  * Copyright 2017 Meniga Iceland Inc.
  */
 public final class MenigaSyncOperationsImp implements MenigaSyncOperations {
 	@Override
-	public Result<MenigaSync> startSync(Long realmUserId, long timeout) {
+	public Result<MenigaSync> startSync(Long realmUserId, @Nullable String sessionToken, long timeout) {
```
Comment (arturdm) [inline sdk/src/main/java/com/meniga/sdk/models/sync/operators/MenigaSyncOperationsImp.java]: I guess `realmUserId` can also be annotated (there's a null check in the method body):

```
public Result<MenigaSync> startSync(@Nullable Long realmUserId, @Nullable String sessionToken, long timeout) {
```

File: sdk/src/main/java/com/meniga/sdk/models/sync/operators/MenigaSyncOperations.java (hunk 1 above)
Comment (arturdm) [reply sdk/src/main/java/com/meniga/sdk/models/sync/operators/MenigaSyncOperations.java]: The signature (+ annotations) can be the same as in `MenigaSyncOperationsImp`

Comment (agustk) [reply sdk/src/main/java/com/meniga/sdk/models/sync/operators/MenigaSyncOperations.java]: Fixed

File: sdk/src/main/java/com/meniga/sdk/models/sync/operators/MenigaSyncOperationsImp.java (hunk 3 above)
Comment (agustk) [reply sdk/src/main/java/com/meniga/sdk/models/sync/operators/MenigaSyncOperationsImp.java]: Added

**Review by arturdm — APPROVED** [review]

**Pull request merged**